src/libcore/hash/mod.rs

```diff
@@ -76,22 +72,123 @@ mod sip;
 /// to compute the hash. Specific implementations of this trait may specialize
 /// for particular instances of `H` in order to be able to optimize the hashing
 /// behavior.
+#[cfg(not(stage0))]
```

Comment:
The doc comment above here is slightly stale -- some of it should move to the methods, I think.